wordpress-mobile/WordPress-iOS
# Post likes: add to Core Data when fetched

Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15662
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/391

This adds a new `PostService+Likes` class that:
- Fetches likes for a Post and returns an array of `LikeUser`s.
  - The new method is basically a Swift version of the [original](https://github.com/wordpress-mobile/WordPress-iOS/blob/8de804c16e55f31cc03da52083bbb7e5b8110983/WordPress/Classes/Services/PostService.m#L688) `PostService` ObjC method, which will be removed later.
- Purges any existing users before creating new ones.

Notes:
- The likes list UI still displays the old `RemoteUser` list.
- The user purging (`deleteExistingUsersFor:postID`) is not filtered. I need to add `postID` to the WPKit response parsing.
- This change is only for Post likes. Comment likes coming soon.

To test:

`LikesListController` still uses the old fetching method to display the likers. I've added a temporary call to the new fetch method that logs out the `LikeUser` array. 

- Go to Notifications > Likes.
- Tap on a Post like notification.
- Verify the user array appears on the console (prefaced by `🔴 likeUsers: `. The users should match the displayed users.


## Regression Notes
1. Potential unintended areas of impact
N/A. Feature incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature incomplete.

PR submission checklist:
